jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/ByteBufferPool.java

```diff
@@ -63,11 +63,9 @@ public interface ByteBufferPool
      * If the buffer is not in a pool, calling this method is equivalent to calling {@link RetainableByteBuffer#release()}.
      * Calling this method satisfies any contract that requires a call to {@link RetainableByteBuffer#release()}.
      * @return {@code true} if a call to {@link RetainableByteBuffer#release()} would have returned {@code true}.
-     * @see RetainableByteBuffer#release()
-     * @deprecated This API is experimental and may be removed in future releases
+     * @see RetainableByteBuffer#releaseForRemoval()
      */
-    @Deprecated
-    default boolean removeAndRelease(RetainableByteBuffer buffer)
+    default boolean releaseForRemoval(RetainableByteBuffer buffer)
```

Comment:
I find the name strange. How about `releaseAndRemove()` or just `remove()` with javadocs explaining that also has release semantic?